Para comenzar, importo los datos de los links correspondientes, tanto los datos de los csv como las codificaciones de categorías del Excel.

In [ ]:
!pip install -q xlrd
import pandas as pd
import altair as alt

In [ ]:
dfprov = pd.read_excel('http://www.deis.msal.gov.ar/wp-content/uploads/2019/01/DescDef1.xlsx','PROVRES')
dfsex = pd.read_excel('http://www.deis.msal.gov.ar/wp-content/uploads/2019/01/DescDef1.xlsx','SEXO')
dfcod = pd.read_excel('http://www.deis.msal.gov.ar/wp-content/uploads/2019/01/DescDef1.xlsx','CODMUER')
dfmat = pd.read_excel('http://www.deis.msal.gov.ar/wp-content/uploads/2019/01/DescDef1.xlsx','MAT')

In [ ]:
dfprov.head()

CODIGO                        VALOR
0       2  Ciudad Aut. de Buenos Aires
1       6                 Buenos Aires
2      10                    Catamarca
3      14                      Córdoba
4      18                   Corrientes

In [ ]:
dfsex

CODIGO            VALOR
0       1            Varón
1       2            Mujer
2       9  Sin especificar

In [ ]:
dfcod.head()

CODIGO                                       VALOR
0    A00                                      Cólera
1    A01             Fiebres tifoidea y paratifoidea
2    A02      Otras infecciones debidas a Salmonella
3    A03                                  Shigelosis
4    A04  Otras infecciones intestinales bacterianas

In [ ]:
dfmat

CODIGO                        VALOR
0      M               Muerte materna
1      T        Muerte materna tardía
2      S  Secuela de causa obstétrica
3   Nulo         No es muerte materna

In [ ]:
df10 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2018/06/DefWeb10.csv', encoding='latin-1')
df11 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2018/06/DefWeb11.csv', encoding='latin-1')
df12 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2018/06/DefWeb12.csv', encoding='latin-1')
df13 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2018/06/DefWeb13.csv', encoding='latin-1')
df14 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2018/06/DefWeb14.csv', encoding='latin-1')
df15 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2018/06/DefWeb15.csv', encoding='latin-1')
df16 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2018/06/DefWeb16.csv', encoding='latin-1')
df17 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2019/01/DefWeb17.csv', encoding='latin-1')
df18 = pd.read_csv('http://www.deis.msal.gov.ar/wp-content/uploads/2020/01/DefWeb18.csv', encoding='latin-1')

In [ ]:
df10.head()

PROVRES  SEXO CAUSA  MAT     GRUPEDAD  CUENTA
0        2     2   J98  NaN   07_30 a 34       2
1       14     2   G60  NaN   04_15 a 19       1
2        6     1   I38  NaN  17_80 y más       1
3       30     2   G12  NaN   14_65 a 69       1
4       14     1   Y20  NaN   16_75 a 79       1

In [ ]:
df18.head()

PROVRES  SEXO CAUSA  MAT     GRUPEDAD  CUENTA
0        2     1   A09  NaN   12_55 a 59       1
1        2     1   A09  NaN  17_80 y más       3
2        2     1   A16  NaN   15_70 a 74       3
3        2     1   A19  NaN   07_30 a 34       1
4        2     1   A41  NaN   05_20 a 24       1

Le agrego la columna **año** a cada data frame con el año correspondiente, para luego juntar todos los data frames en uno.

In [ ]:
df10['AÑO'] = 2010
df11['AÑO'] = 2011
df12['AÑO'] = 2012
df13['AÑO'] = 2013
df14['AÑO'] = 2014
df15['AÑO'] = 2015
df16['AÑO'] = 2016
df17['AÑO'] = 2017
df18['AÑO'] = 2018

df13

PROVRES  SEXO CAUSA  MAT           GRUPEDAD  CUENTA   AÑO
0           90     1   K72  NaN         10_45 a 49       1  2013
1           46     1   V20  NaN         11_50 a 54       1  2013
2           54     2   E46  NaN         15_70 a 74       2  2013
3            6     1   V89  NaN         10_45 a 49       3  2013
4           66     1   J18  NaN         16_75 a 79      38  2013
...        ...   ...   ...  ...                ...     ...   ...
48466       66     1   C95  NaN         14_65 a 69       1  2013
48467       14     2   R99  NaN         13_60 a 64       8  2013
48468       58     2   C67  NaN         16_75 a 79       1  2013
48469        6     1   E14  NaN         10_45 a 49      32  2013
48470       78     1   P26  NaN  01_Menor de 1 año       1  2013

[48471 rows x 7 columns]

Junto todos los dataframes en uno.

In [ ]:
agg = [df10,df11,df12,df13,df14,df15,df16,df17,df18]
dfagg = pd.concat(agg)
dfagg

PROVRES  SEXO CAUSA  MAT            GRUPEDAD  CUENTA   AÑO
0            2     2   J98  NaN          07_30 a 34       2  2010
1           14     2   G60  NaN          04_15 a 19       1  2010
2            6     1   I38  NaN         17_80 y más       1  2010
3           30     2   G12  NaN          14_65 a 69       1  2010
4           14     1   Y20  NaN          16_75 a 79       1  2010
...        ...   ...   ...  ...                 ...     ...   ...
49826       99     2   Y17  NaN          04_15 a 19       1  2018
49827       99     2   Y20  NaN          10_45 a 49       1  2018
49828       99     2   Y34  NaN          06_25 a 29       1  2018
49829       99     2   Y34  NaN          16_75 a 79       1  2018
49830       99     9   I49  NaN  99_Sin especificar       1  2018

[442394 rows x 7 columns]

Creo un dataframe para nombrar mejor las edades.

In [ ]:
data = {'GRUPEDAD':  ['01_Menor de 1 año', '02_1 a 9', '03_10 a 14', '04_15 a 19', '05_20 a 24', '06_25 a 29', '07_30 a 34', '08_35 a 39', '09_40 a 44', '10_45 a 49', '11_50 a 54', '12_55 a 59', '13_60 a 64', '14_65 a 69', '15_70 a 74', '16_75 a 79', '17_80 y más', '99_Sin especificar'],
        'EDAD': ['1-', '1-9', '10-14', '15-19', '20-24', '25-29', '30-34', '35-39', '40-44', '45-49', '50-54', '55-59', '60-64', '65-69', '70-74', '75-79', '80+', 'Sin especificar']
       }
dfedades = pd.DataFrame(data, columns = ['GRUPEDAD', 'EDAD'])
dfedades

GRUPEDAD             EDAD
0    01_Menor de 1 año               1-
1             02_1 a 9              1-9
2           03_10 a 14            10-14
3           04_15 a 19            15-19
4           05_20 a 24            20-24
5           06_25 a 29            25-29
6           07_30 a 34            30-34
7           08_35 a 39            35-39
8           09_40 a 44            40-44
9           10_45 a 49            45-49
10          11_50 a 54            50-54
11          12_55 a 59            55-59
12          13_60 a 64            60-64
13          14_65 a 69            65-69
14          15_70 a 74            70-74
15          16_75 a 79            75-79
16         17_80 y más              80+
17  99_Sin especificar  Sin especificar

Creo un dataframe con la **población de cada provincia** según el censo 2010, para sacar ratios.

In [ ]:
pob = {'PROVINCIA': [2,6,10,22,26,14,18,30,34,38,42,46,50,54,58,62,66,70,74,78,82,86,94,90],
       'POBLACION': [2508373,14683143,366265,1048763,477898,3258388,984949,1225601,509042,643736,315504,330436,1673310,1057581,516611,591391,1186309,675034,426754,247923,3157105,871122,115906,1439995]   
      }

dfcenso = pd.DataFrame(pob, columns = ['PROVINCIA', 'POBLACION'])
dfcenso

PROVINCIA  POBLACION
0           2    2508373
1           6   14683143
2          10     366265
3          22    1048763
4          26     477898
5          14    3258388
6          18     984949
7          30    1225601
8          34     509042
9          38     643736
10         42     315504
11         46     330436
12         50    1673310
13         54    1057581
14         58     516611
15         62     591391
16         66    1186309
17         70     675034
18         74     426754
19         78     247923
20         82    3157105
21         86     871122
22         94     115906
23         90    1439995

Ahora calculemos el número de **defunciones por año** y comparemos **según sexo**.

In [ ]:
#Con esto se puede agregar una columna nueva al dataframe original poniendo el sexo correspondiente a cada número
#No lo voy a hacer así, lo voy a hacer con joins

#import numpy as np
#conditions = [
#    (dfagg['SEXO'] == 1),
#    (dfagg['SEXO'] == 2),
#    (dfagg['SEXO'] == 9)]
#choices = ['Varón', 'Mujer', 'Sin especificar']
#dfagg['SEXO2'] = np.select(conditions, choices)
#dfagg

In [ ]:
#df_bysex = dfagg.groupby(['AÑO','SEXO2'])['CUENTA'].sum().to_frame()
#df_bysex3 = df_bysex.reset_index()
#df_bysex3

In [ ]:
df_bysex2 = dfagg.groupby(['AÑO','SEXO'])['CUENTA'].sum().to_frame().join(dfsex.set_index('CODIGO'), on='SEXO')
df_bysex4 = df_bysex2.reset_index()

new_columns = df_bysex4.columns.values
new_columns[3] = "Sexo"
df_bysex4.columns = new_columns

df_bysex4 = df_bysex4[['AÑO','CUENTA','Sexo']]
df_bysex4

AÑO  CUENTA             Sexo
0   2010  164771            Varón
1   2010  153510            Mujer
2   2010     321  Sin especificar
3   2011  164866            Varón
4   2011  153922            Mujer
5   2011     271  Sin especificar
6   2012  166735            Varón
7   2012  152313            Mujer
8   2012     491  Sin especificar
9   2013  169109            Varón
10  2013  156688            Mujer
11  2013     400  Sin especificar
12  2014  168215            Varón
13  2014  156987            Mujer
14  2014     337  Sin especificar
15  2015  171964            Varón
16  2015  161260            Mujer
17  2015     183  Sin especificar
18  2016  181339            Varón
19  2016  171408            Mujer
20  2016     245  Sin especificar
21  2017  174221            Varón
22  2017  166731            Mujer
23  2017     736  Sin especificar
24  2018  173950            Varón
25  2018  162408            Mujer
26  2018     465  Sin especificar

In [ ]:
alt.Chart(df_bysex4).mark_bar().encode(
    x=alt.X('Sexo:O', 
          title='', 
          axis=alt.Axis(labelAngle=0)),
    y=alt.Y('CUENTA:Q', 
          title='Cantidad de defunciones'),
    color=alt.Color('Sexo:N',
                    scale=alt.Scale(
                      domain=['Varón', 'Mujer'],
                      range=['#4c78a8', '#e45756'])),
    column='AÑO:O'
)

alt.Chart(...)

In [ ]:
base = alt.Chart(df_bysex4).encode(
    x=alt.X('CUENTA:Q', 
            title='Número de defunciones'
            ),
    y=alt.Y('Sexo:O', 
          title="", 
          axis=alt.Axis(labelAngle=0),
          sort='-x'),
    color=alt.Color('Sexo:N',
                    scale=alt.Scale(
                      domain=['Varón', 'Mujer','Sin especificar'],
                      range=['#4c78a8', '#e45756','#f58518']))
)

alt.layer(
  base.mark_bar(),
  base.mark_text(dx=20).encode(text='CUENTA')
).facet(
  row=alt.Row('AÑO:O', sort=['2018','2017','2016','2015','2014','2013','2012','2011','2010'])
)

alt.FacetChart(...)

Ahora veamos el **índice de defunciones por provincia** para 2010 y 2018, teniendo en cuenta el número de defunciones y la población de cada una. Comparemos si hubieron cambios en el tiempo (si bien para ambos índices se tiene en cuenta la población según el censo de 2010).

In [ ]:
dfagg.head()

PROVRES  SEXO CAUSA  MAT     GRUPEDAD  CUENTA   AÑO
0        2     2   J98  NaN   07_30 a 34       2  2010
1       14     2   G60  NaN   04_15 a 19       1  2010
2        6     1   I38  NaN  17_80 y más       1  2010
3       30     2   G12  NaN   14_65 a 69       1  2010
4       14     1   Y20  NaN   16_75 a 79       1  2010

In [ ]:
df_byprov = dfagg.groupby(['AÑO','PROVRES'])['CUENTA'].sum().to_frame().join(dfprov.set_index('CODIGO'), on='PROVRES')
df_byprov2 = df_byprov.join(dfcenso.set_index('PROVINCIA'), on='PROVRES')
df_byprov3 = df_byprov2.reset_index()

new_columns2 = df_byprov3.columns.values
new_columns2[3] = "Provincia"
df_byprov3.columns = new_columns2

df_byprov3['RATIO'] = df_byprov3['CUENTA']/df_byprov3['POBLACION']*100

df_byprov4 = df_byprov3[['AÑO','CUENTA','Provincia','POBLACION','RATIO']]
df_byprov4 = df_byprov4.drop(df_byprov4[df_byprov4.Provincia == "Otro país"].index)
df_byprov4 = df_byprov4.drop(df_byprov4[df_byprov4.Provincia == "Lugar no especificado"].index)

df_byprov4

AÑO  CUENTA                    Provincia   POBLACION     RATIO
0    2010   33111  Ciudad Aut. de Buenos Aires   2508373.0  1.320019
1    2010  129403                 Buenos Aires  14683143.0  0.881303
2    2010    2278                    Catamarca    366265.0  0.621954
3    2010   27874                      Córdoba   3258388.0  0.855454
4    2010    6489                   Corrientes    984949.0  0.658816
..    ...     ...                          ...         ...       ...
227  2018    1641                   Santa Cruz    247923.0  0.661899
228  2018   29382                     Santa Fe   3157105.0  0.930663
229  2018    6163          Santiago del Estero    871122.0  0.707478
230  2018   11101                      Tucumán   1439995.0  0.770905
231  2018     603             Tierra del Fuego    115906.0  0.520249

[216 rows x 5 columns]

In [ ]:
#2010
df_byprov5 = df_byprov4.loc[df_byprov4['AÑO'] == 2010]

alt.Chart(df_byprov5).mark_bar(color='#72b7b2').encode(
    x=alt.X('RATIO:Q', 
          title='Índice de defunciones por cantidad de habitantes'),
    y=alt.Y('Provincia:N', 
          title='Provincia', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
)

alt.Chart(...)

Vemos que en 2010 las 3 provincias con mayor índice de defunciones por cantidad de habitantes fueron **la Ciudad de Buenos Aires** (aprox 1.3%), **Santa Fe** (aprox 0.9%) y **la provincia de Buenos Aires** (aprox 0.9%).

In [ ]:
#2018
df_byprov6 = df_byprov4.loc[df_byprov4['AÑO'] == 2018]

alt.Chart(df_byprov6).mark_bar(color='#72b7b2').encode(
    x=alt.X('RATIO:Q', 
          title='Índice de defunciones por cantidad de habitantes'),
    y=alt.Y('Provincia:N', 
          title='Provincia', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
)

alt.Chart(...)

Vemos que en 2018 las 3 provincias con mayor índice de defunciones por cantidad de habitantes se mantuvieron iguales a las de 2010, siendo estas **la Ciudad de Buenos Aires** (aprox 1.2%), **Santa Fe** (aprox 0.9%) y **la provincia de Buenos Aires** (aprox 0.9%). De todos modos, vemos un crecimiento en los índices de defunciones de casi todas las provincias.

Ahora analicemos las **principales causas de defunción**, tanto para hombres como para mujeres. Tomaremos las 10 más comunes para cada uno.

In [ ]:
dfsex

CODIGO            VALOR
0       1            Varón
1       2            Mujer
2       9  Sin especificar

Primero saco la cuenta de cantidad total de varones, mujeres y personas sin especificar fallecidos.

In [ ]:
df_cantdef = dfagg.groupby(['SEXO'])['CUENTA'].sum().to_frame().join(dfsex.set_index('CODIGO'), on='SEXO').reset_index()
df_cantdef.rename(columns = {'VALOR':'Sexo'}, inplace = True)
df_cantdef = df_cantdef[['CUENTA','Sexo']]
df_cantdef

CUENTA             Sexo
0  1535170            Varón
1  1435227            Mujer
2     3449  Sin especificar

Ahora agrupo las muertes por sexo y causa.

In [ ]:
df_bycause = dfagg.groupby(['SEXO','CAUSA'])['CUENTA'].sum().to_frame().join(dfcod.set_index('CODIGO'), on='CAUSA')
df_bycause = df_bycause.reset_index()

new_columns3 = df_bycause.columns.values
new_columns3[3] = "Causa de muerte"
df_bycause.columns = new_columns3

df_bycause2 = df_bycause.join(dfsex.set_index('CODIGO'), on='SEXO')

new_columns4 = df_bycause2.columns.values
new_columns4[4] = "Sexo"
df_bycause2.columns = new_columns4

df_bycause3 = df_bycause2[['Sexo','Causa de muerte','CUENTA']]
df_bycause3

Sexo  ... CUENTA
0               Varón  ...      2
1               Varón  ...      7
2               Varón  ...      3
3               Varón  ...    123
4               Varón  ...    120
...               ...  ...    ...
2817  Sin especificar  ...     15
2818  Sin especificar  ...      7
2819  Sin especificar  ...      9
2820  Sin especificar  ...     38
2821  Sin especificar  ...      3

[2822 rows x 3 columns]

Ahora tomo las 10 causas más comunes de muerte en varones, sacando el porcentaje del total de varones que fallecieron debido a la misma.

In [ ]:
#Varones
df_bycause4 = df_bycause3.loc[df_bycause3['Sexo'] == 'Varón']
df_bycause4 = df_bycause4.sort_values(by='CUENTA', ascending=False)
df_bycause4 = df_bycause4.head(10)
df_bycause4['PORCENTAJE'] = df_bycause4['CUENTA']/1535170*100
df_bycause4

Sexo  ... PORCENTAJE
451  Varón  ...   7.362963
494  Varón  ...   6.698216
428  Varón  ...   5.589023
907  Varón  ...   4.524841
140  Varón  ...   3.745904
27   Varón  ...   2.869129
533  Varón  ...   2.860270
457  Varón  ...   2.522848
155  Varón  ...   2.224770
509  Varón  ...   2.098139

[10 rows x 4 columns]

In [ ]:
#Varones
alt.Chart(df_bycause4).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
)

alt.Chart(...)

Vemos que la causa de defunción más común en varones es **insuficiencia cardíaca** (7.36%), seguida de **neumonía organismo no especificado** (6.70%) e **infarto agudo del miocardio** (5.59%).

In [ ]:
#Mujeres
df_bycause5 = df_bycause3.loc[df_bycause3['Sexo'] == 'Mujer']
df_bycause5 = df_bycause5.sort_values(by='CUENTA', ascending=False)
df_bycause5 = df_bycause5.head(10)
df_bycause5['PORCENTAJE'] = df_bycause5['CUENTA']/1435227*100

df_bycause5

Sexo  ... PORCENTAJE
1696  Mujer  ...   9.771695
1738  Mujer  ...   8.115650
2218  Mujer  ...   4.823279
1673  Mujer  ...   4.307681
1776  Mujer  ...   3.672938
1388  Mujer  ...   3.580200
1267  Mujer  ...   3.471855
1702  Mujer  ...   2.873413
1513  Mujer  ...   1.966030
1375  Mujer  ...   1.851554

[10 rows x 4 columns]

In [ ]:
#Mujeres
alt.Chart(df_bycause5).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
)

alt.Chart(...)

Vemos que la causa de defunción más común en mujeres es **insuficiencia cardíaca** (9.77%), seguida de **neumonía organismo no especificado** (8.12%) y **otras causas mal definidas y las no especificadas** (4.82%). Las dos primeras son las mismas que en el caso de los varones, aunque ambas son más frecuentes en mujeres. La tercera es distinta, aunque es la 4ta más frecuente en varones, al igual que la 3ra más frecuente en varones es la 4ta más frecuente en mujeres (están invertidas).

Cabe destacar que algunas causas de muerte difieren bastante según el sexo, como la 5ta causa de muerte más común en hombres, que es **tumor maligno de los bronquios y del pulmón**, bastante menos frecuente en mujeres (3.75% versus 1.85%), o la 6ta causa más frecuente en mujeres, inexistente en varones, que es **tumor maligno de la mama** (3.58%).

Por último veamos las causas de muerte más frecuentes en individuos cuyo sexo no está especificado, con el fin de analizar si el sexo del cuerpo era indistinguible al momento de determinar la causa de muerte (por ejemplo por quemaduras, desmembramiento, o similar), o si simplemente no se registró el sexo del individuo.

In [ ]:
#Sin especificar
df_bycause6 = df_bycause3.loc[df_bycause3['Sexo'] == 'Sin especificar']
df_bycause6 = df_bycause6.sort_values(by='CUENTA', ascending=False)
df_bycause6 = df_bycause6.head(10)
df_bycause6['PORCENTAJE'] = df_bycause6['CUENTA']/3449*100

df_bycause6

Sexo  ... PORCENTAJE
2772  Sin especificar  ...   8.843143
2646  Sin especificar  ...   8.437228
2668  Sin especificar  ...   6.320673
2681  Sin especificar  ...   5.044941
2630  Sin especificar  ...   4.494056
2527  Sin especificar  ...   3.305306
2651  Sin especificar  ...   2.435489
2553  Sin especificar  ...   2.029574
2598  Sin especificar  ...   1.710641
2780  Sin especificar  ...   1.681647

[10 rows x 4 columns]

In [ ]:
#Sin especificar
alt.Chart(df_bycause6).mark_bar(color='#f58518').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
)

alt.Chart(...)

Vemos que las principales causas de muerte son similares a las de varones y mujeres, lo cual sugiere que simplemente el registro del sexo fue deficiente al momento de la muerte.

Ahora comparemos las 5 principales causas de muerte para varones y mujeres, en cada año.

In [ ]:
df_cantdef2 = dfagg.groupby(['AÑO','SEXO'])['CUENTA'].sum().to_frame().join(dfsex.set_index('CODIGO'), on='SEXO').reset_index()
df_cantdef2.rename(columns = {'VALOR':'Sexo'}, inplace = True)
df_cantdef2 = df_cantdef2[['AÑO','Sexo','CUENTA']]
df_cantdef2

AÑO             Sexo  CUENTA
0   2010            Varón  164771
1   2010            Mujer  153510
2   2010  Sin especificar     321
3   2011            Varón  164866
4   2011            Mujer  153922
5   2011  Sin especificar     271
6   2012            Varón  166735
7   2012            Mujer  152313
8   2012  Sin especificar     491
9   2013            Varón  169109
10  2013            Mujer  156688
11  2013  Sin especificar     400
12  2014            Varón  168215
13  2014            Mujer  156987
14  2014  Sin especificar     337
15  2015            Varón  171964
16  2015            Mujer  161260
17  2015  Sin especificar     183
18  2016            Varón  181339
19  2016            Mujer  171408
20  2016  Sin especificar     245
21  2017            Varón  174221
22  2017            Mujer  166731
23  2017  Sin especificar     736
24  2018            Varón  173950
25  2018            Mujer  162408
26  2018  Sin especificar     465

In [ ]:
df_bycause7 = dfagg.groupby(['AÑO','SEXO','CAUSA'])['CUENTA'].sum().to_frame().join(dfcod.set_index('CODIGO'), on='CAUSA')
df_bycause7 = df_bycause7.reset_index()

new_cols = df_bycause7.columns.values
new_cols[4] = "Causa de muerte"
df_bycause7.columns = new_cols

df_bycause8 = df_bycause7.join(dfsex.set_index('CODIGO'), on='SEXO')

new_cols2 = df_bycause8.columns.values
new_cols2[5] = "Sexo"
df_bycause8.columns = new_cols2

df_bycause9 = df_bycause8[['AÑO','Sexo','Causa de muerte','CUENTA']]
df_bycause9

AÑO  ... CUENTA
0      2010  ...      2
1      2010  ...      1
2      2010  ...     15
3      2010  ...      1
4      2010  ...    147
...     ...  ...    ...
17084  2018  ...      1
17085  2018  ...      2
17086  2018  ...      1
17087  2018  ...      2
17088  2018  ...      7

[17089 rows x 4 columns]

In [ ]:
#Varones

df_bycause10 = df_bycause9.loc[df_bycause9['Sexo'] == 'Varón']
#df_bycause10['PORCENTAJE'] = df_bycause10['CUENTA']/df_cantdef2['CUENTA']

df_bycause10.loc[df_bycause10['AÑO'] == 2010, 'PORCENTAJE'] = df_bycause10['CUENTA']/164771*100
df_bycause10.loc[df_bycause10['AÑO'] == 2011, 'PORCENTAJE'] = df_bycause10['CUENTA']/164866*100
df_bycause10.loc[df_bycause10['AÑO'] == 2012, 'PORCENTAJE'] = df_bycause10['CUENTA']/166735*100
df_bycause10.loc[df_bycause10['AÑO'] == 2013, 'PORCENTAJE'] = df_bycause10['CUENTA']/169109*100
df_bycause10.loc[df_bycause10['AÑO'] == 2014, 'PORCENTAJE'] = df_bycause10['CUENTA']/168215*100
df_bycause10.loc[df_bycause10['AÑO'] == 2015, 'PORCENTAJE'] = df_bycause10['CUENTA']/171964*100
df_bycause10.loc[df_bycause10['AÑO'] == 2016, 'PORCENTAJE'] = df_bycause10['CUENTA']/181339*100
df_bycause10.loc[df_bycause10['AÑO'] == 2017, 'PORCENTAJE'] = df_bycause10['CUENTA']/174221*100
df_bycause10.loc[df_bycause10['AÑO'] == 2018, 'PORCENTAJE'] = df_bycause10['CUENTA']/173950*100

df_bycause10

AÑO   Sexo  ... CUENTA  PORCENTAJE
0      2010  Varón  ...      2    0.001214
1      2010  Varón  ...      1    0.000607
2      2010  Varón  ...     15    0.009104
3      2010  Varón  ...      1    0.000607
4      2010  Varón  ...    147    0.089215
...     ...    ...  ...    ...         ...
16070  2018  Varón  ...      3    0.001725
16071  2018  Varón  ...      3    0.001725
16072  2018  Varón  ...      1    0.000575
16073  2018  Varón  ...      2    0.001150
16074  2018  Varón  ...      1    0.000575

[8147 rows x 5 columns]

In [ ]:
#Varones

df_bycause_v_sub1 = df_bycause10.loc[df_bycause10['AÑO'] == 2010]
df_bycause_v_sub1 = df_bycause_v_sub1.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub1 = df_bycause_v_sub1.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub2 = df_bycause10.loc[df_bycause10['AÑO'] == 2011]
df_bycause_v_sub2 = df_bycause_v_sub2.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub2 = df_bycause_v_sub2.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub3 = df_bycause10.loc[df_bycause10['AÑO'] == 2012]
df_bycause_v_sub3 = df_bycause_v_sub3.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub3 = df_bycause_v_sub3.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub4 = df_bycause10.loc[df_bycause10['AÑO'] == 2013]
df_bycause_v_sub4 = df_bycause_v_sub4.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub4 = df_bycause_v_sub4.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub5 = df_bycause10.loc[df_bycause10['AÑO'] == 2014]
df_bycause_v_sub5 = df_bycause_v_sub5.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub5 = df_bycause_v_sub5.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub6 = df_bycause10.loc[df_bycause10['AÑO'] == 2015]
df_bycause_v_sub6 = df_bycause_v_sub6.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub6 = df_bycause_v_sub6.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub7 = df_bycause10.loc[df_bycause10['AÑO'] == 2016]
df_bycause_v_sub7 = df_bycause_v_sub7.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub7 = df_bycause_v_sub7.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub8 = df_bycause10.loc[df_bycause10['AÑO'] == 2017]
df_bycause_v_sub8 = df_bycause_v_sub8.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub8 = df_bycause_v_sub8.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub9 = df_bycause10.loc[df_bycause10['AÑO'] == 2018]
df_bycause_v_sub9 = df_bycause_v_sub9.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_v_sub9 = df_bycause_v_sub9.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_v_sub3

Causa de muerte  PORCENTAJE
0                             Insuficiencia cardíaca    8.619066
1                        Infarto agudo del miocardio    5.456563
2                 Neumonía organismo no especificado    5.413381
3  Otras causas mal definidas y las no especifica...    5.154287
4        Tumor maligno de los bronquios y del pulmón    3.879210

In [ ]:
#Mujeres

df_bycause11 = df_bycause9.loc[df_bycause9['Sexo'] == 'Mujer']
#df_bycause10['PORCENTAJE'] = df_bycause10['CUENTA']/df_cantdef2['CUENTA']

df_bycause11.loc[df_bycause11['AÑO'] == 2010, 'PORCENTAJE'] = df_bycause11['CUENTA']/153510*100
df_bycause11.loc[df_bycause11['AÑO'] == 2011, 'PORCENTAJE'] = df_bycause11['CUENTA']/153922*100
df_bycause11.loc[df_bycause11['AÑO'] == 2012, 'PORCENTAJE'] = df_bycause11['CUENTA']/152313*100
df_bycause11.loc[df_bycause11['AÑO'] == 2013, 'PORCENTAJE'] = df_bycause11['CUENTA']/156688*100
df_bycause11.loc[df_bycause11['AÑO'] == 2014, 'PORCENTAJE'] = df_bycause11['CUENTA']/156987*100
df_bycause11.loc[df_bycause11['AÑO'] == 2015, 'PORCENTAJE'] = df_bycause11['CUENTA']/161260*100
df_bycause11.loc[df_bycause11['AÑO'] == 2016, 'PORCENTAJE'] = df_bycause11['CUENTA']/171408*100
df_bycause11.loc[df_bycause11['AÑO'] == 2017, 'PORCENTAJE'] = df_bycause11['CUENTA']/166731*100
df_bycause11.loc[df_bycause11['AÑO'] == 2018, 'PORCENTAJE'] = df_bycause11['CUENTA']/162408*100

df_bycause11

AÑO   Sexo  ... CUENTA  PORCENTAJE
901    2010  Mujer  ...      2    0.001303
902    2010  Mujer  ...      1    0.000651
903    2010  Mujer  ...     11    0.007166
904    2010  Mujer  ...      2    0.001303
905    2010  Mujer  ...      2    0.001303
...     ...    ...  ...    ...         ...
16968  2018  Mujer  ...    466    0.286932
16969  2018  Mujer  ...     26    0.016009
16970  2018  Mujer  ...      2    0.001231
16971  2018  Mujer  ...      3    0.001847
16972  2018  Mujer  ...      1    0.000616

[7959 rows x 5 columns]

In [ ]:
df_bycause_m_sub1 = df_bycause11.loc[df_bycause11['AÑO'] == 2010]
df_bycause_m_sub1 = df_bycause_m_sub1.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub1 = df_bycause_m_sub1.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub2 = df_bycause11.loc[df_bycause11['AÑO'] == 2011]
df_bycause_m_sub2 = df_bycause_m_sub2.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub2 = df_bycause_m_sub2.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub3 = df_bycause11.loc[df_bycause11['AÑO'] == 2012]
df_bycause_m_sub3 = df_bycause_m_sub3.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub3 = df_bycause_m_sub3.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub4 = df_bycause11.loc[df_bycause11['AÑO'] == 2013]
df_bycause_m_sub4 = df_bycause_m_sub4.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub4 = df_bycause_m_sub4.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub5 = df_bycause11.loc[df_bycause11['AÑO'] == 2014]
df_bycause_m_sub5 = df_bycause_m_sub5.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub5 = df_bycause_m_sub5.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub6 = df_bycause11.loc[df_bycause11['AÑO'] == 2015]
df_bycause_m_sub6 = df_bycause_m_sub6.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub6 = df_bycause_m_sub6.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub7 = df_bycause11.loc[df_bycause11['AÑO'] == 2016]
df_bycause_m_sub7 = df_bycause_m_sub7.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub7 = df_bycause_m_sub7.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub8 = df_bycause11.loc[df_bycause11['AÑO'] == 2017]
df_bycause_m_sub8 = df_bycause_m_sub8.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub8 = df_bycause_m_sub8.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub9 = df_bycause11.loc[df_bycause11['AÑO'] == 2018]
df_bycause_m_sub9 = df_bycause_m_sub9.sort_values(by='PORCENTAJE', ascending=False)
df_bycause_m_sub9 = df_bycause_m_sub9.reset_index()[['Causa de muerte','PORCENTAJE']].head(5)

df_bycause_m_sub9

Causa de muerte  PORCENTAJE
0                 Neumonía organismo no especificado    9.666396
1                             Insuficiencia cardíaca    7.732993
2                        Infarto agudo del miocardio    4.621078
3  Otras causas mal definidas y las no especifica...    3.861263
4  Insuficiencia respiratoria no clasificada en o...    3.569405

In [ ]:
chart_v_1 = alt.Chart(df_bycause_v_sub1).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2010'
)

chart_m_1 = alt.Chart(df_bycause_m_sub1).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2010'
)

chart_v_2 = alt.Chart(df_bycause_v_sub2).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2011'
)

chart_m_2 = alt.Chart(df_bycause_m_sub2).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2011'
)

chart_v_3 = alt.Chart(df_bycause_v_sub3).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2012'
)

chart_m_3 = alt.Chart(df_bycause_m_sub3).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2012'
)

chart_v_4 = alt.Chart(df_bycause_v_sub4).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2013'
)

chart_m_4 = alt.Chart(df_bycause_m_sub4).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2013'
)

chart_v_5 = alt.Chart(df_bycause_v_sub5).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2014'
)

chart_m_5 = alt.Chart(df_bycause_m_sub5).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2014'
)

chart_v_6 = alt.Chart(df_bycause_v_sub6).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2015'
)

chart_m_6 = alt.Chart(df_bycause_m_sub6).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2015'
)

chart_v_7 = alt.Chart(df_bycause_v_sub7).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2016'
)

chart_m_7 = alt.Chart(df_bycause_m_sub7).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2016'
)

chart_v_8 = alt.Chart(df_bycause_v_sub8).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2017'
)

chart_m_8 = alt.Chart(df_bycause_m_sub8).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2017'
)

chart_v_9 = alt.Chart(df_bycause_v_sub9).mark_bar(color='#4c78a8').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2018'
)

chart_m_9 = alt.Chart(df_bycause_m_sub9).mark_bar(color='#e45756').encode(
    x=alt.X('PORCENTAJE:Q', 
          title='Porcentaje'),
    y=alt.Y('Causa de muerte:N', 
          title='Causa de muerte', 
          axis=alt.Axis(labelAngle=0),
          sort='-x')
).properties(
    width=300,
    title='2018'
)

(chart_v_1 & chart_v_2 & chart_v_3 & chart_v_4 & chart_v_5 & chart_v_6 & chart_v_7 & chart_v_8 & chart_v_9) | (chart_m_1 & chart_m_2 & chart_m_3 & chart_m_4 & chart_m_5 & chart_m_6 & chart_m_7 & chart_m_8 & chart_m_9)

alt.HConcatChart(...)

Podemos ver que tanto para varones como para mujeres, la principal causa de muerte cambió con el tiempo, pasando de ser **insuficiencia cardíaca** entre 2010 y 2014 en el caso de los varones, y 2010 y 2015 en el caso de las mujeres, a ser **neumonía organismo no especificado**.

También podemos observar que una de las 5 principales causas de muerte en los hombres en todos los años es **tumor maligno de los bronquios y del pulmón** mientras que para las mujeres no lo es en ningún año, aunque sí lo es, de 2015 en adelante, **tumor maligno de la mama**.